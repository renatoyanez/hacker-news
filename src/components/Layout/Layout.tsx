import { ReactElement, ReactNode } from 'react';
import { Header, Typography, Tabs } from 'components';
import { RouteList } from 'routes';
import styles from './layout.module.css';

export interface LayoutProps {
	children?: ReactNode | ReactNode[];
}

const Layout = ({ children }: LayoutProps): ReactElement => {
	return (
		<div>
			<Header title={<Typography>HACKER NEWS</Typography>} />
			<div className={styles.layout}>
				<div style={{ marginBottom: '4.375rem ' }}>
					<Tabs
						data={[
							{ title: 'All', path: RouteList.HOME },
							{ title: 'My faves', path: RouteList.FAVES },
						]}
					/>
				</div>
				{children}
			</div>
		</div>
	);
};

export default Layout;
