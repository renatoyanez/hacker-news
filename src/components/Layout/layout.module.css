.layout {
	display: flex;
	flex-direction: column;
	align-items: center;
	padding: 4.375rem 9.375rem 6.125rem 9.375rem;
}
