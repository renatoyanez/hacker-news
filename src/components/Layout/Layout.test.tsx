import { ReactNode } from 'react';
import { create } from 'react-test-renderer';
import { Router as MemRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Layout from './Layout';

describe('Layout', () => {
	const child = [] as ReactNode;
	const history = createMemoryHistory();

	it('renders correctly', () => {
		const tree = create(
			<MemRouter location={history.location} navigator={history}>
				<Layout>{child}</Layout>
			</MemRouter>,
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
