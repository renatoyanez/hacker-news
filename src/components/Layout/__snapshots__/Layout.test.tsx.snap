// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <p
      className="textStyle"
    >
      HACKER NEWS
    </p>
  </div>
  <div
    className="layout"
  >
    <div
      style={
        Object {
          "marginBottom": "4.375rem ",
        }
      }
    >
      <div>
        <button
          className="selectedButton first"
          onClick={[Function]}
        >
          <p
            className="textStyle2"
          >
            All
          </p>
        </button>
        <button
          className="button second"
          onClick={[Function]}
        >
          <p
            className="textStyle5"
          >
            My faves
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
